docs/dev/running-test.md

```diff
@@ -1,64 +1,97 @@
+
+This section will show you how you can deploy the assisted installer to
+a minikube cluster in preperation for subsystem tests.
+
+### minikube
+
+First we must prepare the minikube cluster -
 
 ```bash
+# Optionally delete the existing minikube cluster:
+# minikube delete
+
 minikube start
+
+# Start the minikube container registry and make it accessible locally:
 minikube addons enable registry
-```
+nohup kubectl port-forward svc/registry 5000:80 -n kube-system &>/dev/null &
+export LOCAL_SUBSYSTEM_REGISTRY=localhost:5000
 
-* Set `LOCAL_ASSISTED_ORG` to point to your local registry address
+sleep 5
 
-```bash
-export LOCAL_ASSISTED_ORG=localhost:5000
+while ! curl --location $LOCAL_SUBSYSTEM_REGISTRY; do
+	echo "kubectl registry service tunnel at port 5000 is not available yet, retrying..."
+    echo "If this persists, try running the kubectl port-forward command above without"
+    echo "nohup, /dev/null redirection and the background job & operator and see if there"
+    echo "are any errors"
+    sleep 3
+done
+
+# Make a tunnel to make minikube services reachable (the command will ask for root password):
+nohup minikube tunnel &>/dev/null &
 ```
 
-* Redirect the local registry to the minikube registry:
+Now that the cluster is prepared, we can deploy the service - 
+
+To deploy the service in REST-API mode, run:
 
 ```bash
-nohup kubectl port-forward svc/registry 5000:80 -n kube-system &>/dev/null &
+skipper make deploy-service-for-subsystem-test
 ```
 
-* Make a tunnel to make minikube services reachable (the command will ask for root password):
+To deploy the service in kube-api mode, run:
 
 ```bash
-nohup minikube tunnel &>/dev/null &
+ENABLE_KUBE_API=true skipper make deploy-service-for-subsystem-test
+skipper make enable-kube-api-for-subsystem
 ```
 
-* Deploy services:
+## Running the subsystem tests
+
+To run the REST-API subsystem tests, run:
 
 ```bash
-skipper make deploy-test
+skipper make subsystem-test
 ```
 
-## Run tests
+To run the kube-api subsystem tests, run:
 
 ```bash
-skipper make [test|unit-test] [environment variables]
+skipper make subsystem-test-kube-api
 ```
 
-* `test` - Runs subsystem tests.
-* `unit-test` - Runs unit tests.
-
-Environment variables:
+Optionally the following environment variables can be exported:
 
-* `FOCUS="install_cluster"` - An optional flag used for [focused specs](https://onsi.github.io/ginkgo/#focused-specs) with regular expression.
-* `SKIP="install_cluster"` - An optional flag to skip scopes with regular expressions.
-* `TEST="./internal/host"` -  An optional flag used for testing a specific package.
-* `VERBOSE=true` - An optional flag to print verbosed data.
+* FOCUS="install_cluster" - An optional flag used for [focused specs](https://onsi.github.io/ginkgo/#focused-specs) with regular expression.
+* SKIP="install_cluster" - An optional flag to skip scopes with regular expressions.
+* TEST="./internal/host" -  An optional flag used for testing a specific package.
```

Comment:
doesn't look related to subsystem flow, it may be confusing no?

Reply:
Right